keps/sig-node/4580-deprecate-kubelet-runonce/README.md

```diff
@@ -0,0 +1,354 @@
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- N/A
+
+##### Integration tests
+
+- N/A
+
+##### e2e tests
+
+- N/A
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature gate `LegacyNodeRunOnceMode` is introduced, is enable by default. Disable this feature gate will fail the kubelet on startup with Runonce mode enable.
+- Mark the `RunOnce` field in `KubeletConfiguration` as deprecated.
+
+#### Beta
+
+- `LegacyNodeRunOnceMode` feature gate is disable by default.
+- Failed when starting kubelet in Runonce mode.
+
+#### GA
+
+- We make the `LegacyNodeRunOnceMode` feature gate disable by default and cannot be enable.
+- Comment the `RunOnce` field in KubeletConfiguration as 'no longer has any effect', and remove the kubelet's `runonce` flag.
+- Remove kubelet Runonce mode.
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
+
+- N/A
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: LegacyNodeRunOnceMode
+  - Components depending on the feature gate: kubelet
+  - Will enabling / disabling the feature require downtime of the control
+    plane? Yes. Flag must be set on kubelet start. To disable, kubelet must be restarted. Hence, there would be brief control component downtime on a given node.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? Yes. See above; disabling would require brief node downtime.
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Using the feature gate is the only way to enable/disable this feature.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+N/A
+
+###### Are there any tests for feature enablement/disablement?
+
+N/A
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+In the alpha `stage`, this feature is enable by default.
+
+Cluster operators can test the behavior by enabling the feature gate.
+
+In the beta `stage`, this feature is enable by default. With this feature disable, the kubelet will refuse to start if is still using Runonce mode.
+
+Cluster operators can reinstate the mode by explicitly enabling the feature gate.
+
+###### What specific metrics should inform a rollback?
+
+N/A
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+N/A
```

Comment:
You are removing flags and configuration fields, so you need to fill this one in.